### ISSUE TITLE
[Low Vis] Failure of Success Criterion 1.4.3 and 1.4.6 due to using background color that do not provide sufficient contrast with foreground text color 

### ISSUE THREAD

**Issue by joshueoconnor** [w3c/wcag#95]
Submitter's Name: aurelien levy
Submitter's Email: aurelien.levy@temesis.com

This failure occurs when people with low vision are not able to read text that is displayed over a background color. When there is not sufficient contrast between the background color and the text color, features of the background color can be confused with the text color making it difficult to accurately read the text.
To satisfy Success Criterion 1.4.3 and 1.4.6, there must be sufficient contrast between the text color and its background color.

Example 1 Head: Failure Example 1
Example 1 Description:
Black text overlays a dark grey background

h2{
color:#000000;
background: #515151; 
}

Example 2 Head: Failure Example 2
Example 2 Description:
Black text overlays a background with a css vertical gradient from black to white.

h2{
color:#000000;
background: linear-gradient(to bottom, #FFFFFF 0%,#000000 100%); 
}

Test Procedure:
1.Quickcheck: First do a quick check to see if the contrast between the text and the area of the background color that is darkest (for dark text) or lightest (for light text) meets or exceeds that required by the Success Criterion (1.4.3 or 1.4.6). If the contrast meets or exceeds the specified contrast, then there is no failure. 
2.If the Quickcheck is false, then check to see if the background color behind each letter has sufficient contrast with the letter.

Expected Result:
If Quickcheck is false and #2 is false as well then this failure condition applies and the content fails the contrast Success Criterion.

Additional Notes:
this failure relates to 1.4.3 and 1.4.6

**Comment by joshueoconnor** [comment]
Related to https://github.com/w3c/wcag/issues/94

**Comment by WayneEDick** [comment]
This should be formalized as a technique relating to gradient backgrounds.  It is exactly what I advise developers to use.

**Comment by WayneEDick** [comment]
This is wording verbatim from Katie Haritos-Shea. I propose it as a technique:
"Text Abbreviation Legend:

·         CCA = Color Contrast Analyzer

·         FG = Foreground

·         BG = Background

 

Primary Test Method for REGULAR Size Text for 1.4.3:

·         Test the REGULAR Size Text values with CCA using this basic method: Use browser Zoom to enlarge the sample area. Open CCA. Sample size must be FG 1 x 1, BG 2 X 2.  Locate the visually apparent lightest and darkest BG colors relative to the FG color, using a manual squint test – or take a sample from 4 locations at North, South, West and East.  Use CCA to test those 2 or 4 areas. Pick the result with the lowest ratio to determine if the ratio passes or fails.

NOTE: If BG is in transition, take sample for BG on the logical stopping points. Take screen capture if there are no stopping points, and use that image for your samples.

·         Take the samples from locations that are specifically chosen to test the foreground color of the text against the background precisely at a place where the lowest degree of contrast exists between the two. Contrast ratio must meet the minimum contrast requirement for Regular Text of 4.5:1.

 

Primary Test Method for LARGE Size Text for 1.4.3:

·         Test the LARGE Size Text values with CCA using this basic method: Use browser Zoom to enlarge the sample area. Open CCA. Sample size must be FG 1 x 1, BG 2 X 2.  Locate the visually apparent lightest and darkest BG colors relative to the FG color, using a manual squint test – or take a sample from 4 locations at North, South, West and East.  Use CCA to test those 2 or 4 areas. Pick the result with the lowest ratio to determine if the ratio passes or fails.
NOTE: If BG is in transition, take sample for BG on the logical stopping points. Take screen capture if there are no stopping points, and use capture for your samples.

·         Take the samples from locations that are specifically chosen to test the foreground color of the text against the background precisely at a place where the lowest degree of contrast exists between the two. Contrast ratio must meet the minimum contrast requirement for Large Text of 3:1.

Test for both REGULAR and LARGE size text should be completed for these scenarios:

·         Text on Solid Background

·         Text on Gradient Background

·         Text on Image Background

·         Text on Scrolling / Animated Background

·         Contrast of Link Text

·         Contrast of Placeholder Text in Forms"

Katie also suggests looking at:

·         Contrast of Images-of-Text

·         Contrast of Visual Focus Indicators

·         Contrast of Caption Text

I agree that these should have contrast thresholds, but the do exceed 1.4.3.  Perhaps  create an Advisory Technique referring to the method given in the new technique.  Give the advice that Visual Focus should use a small print level.

**Comment by joshueoconnor** [comment]
Working group surveyed on these suggestions https://www.w3.org/2002/09/wbs/35422/Sept1st2015/

**Comment by joshueoconnor** [comment]
Discussed by the working group: RESOLUTION: Leave open, until Wayne is free to discuss.

**Comment by joshueoconnor** [comment]
@WayneEDick Do you have any update on this item? Or will you be able to join a call soon to discuss? Thanks

**Comment by WobleDuck** [comment]
I have no more comments on this.  It should be ready for discussion.

**Comment by goetsu** [comment]
any news on this one ?

**Comment by goetsu** [comment]
anyone ?  @michael-n-cooper ?

**Comment by WayneEDick** [comment]
I am placing this on the list of technical problems for the low vision task force.  This is complex enough to wait for more results from the task force.  The basic issue with Kathie's analysis, but the task force may have more to say about complex figures.

**Comment by joshueoconnor** [comment]
Thanks @WayneEDick - we look forward to results from the LVT on this.

**Comment by WayneEDick** [comment]
Eric.
I posted this a use case with the LVTF. There are many threads as Michael C
pointed out, and all have an influence on LV. So, it is tabled for now.
Wayne

**Comment by goetsu** [comment]
any news ?

**Comment by awkawk** [comment]
Preparing to mark as deferred due to inactivity and group resource constraints (https://www.w3.org/WAI/GL/wiki/Handling_Issues).
